discovery_engine_core/providers/src/client.rs

```diff
@@ -219,10 +199,17 @@ impl Client {
     }
 
     /// Run a query for fetching `Article`s from Newscatcher.
-    pub async fn query_articles(&self, query: &impl Query) -> Result<Vec<Article>, Error> {
-        self.query_newscatcher(query)
+    pub async fn query_articles(&self, query: &impl Query) -> Result<Vec<GenericArticle>, Error> {
+        let articles = self
+            .query_newscatcher(query)
             .await
-            .map(|news| news.articles)
+            .map(|news| news.articles)?;
+
+        let articles: Vec<GenericArticle> = articles
+            .into_iter()
+            .flat_map(GenericArticle::try_from)
```

Comment:
🔧  I'm not sure about this solution because it still produces a different behaviour than we currently have. 
Currently, we log and collect the conversion errors in `documentify_articles`
https://github.com/xaynetwork/xayn_discovery_engine/blob/126a0b034d5c7aa0ade8d30d6e75f108fda96933/discovery_engine_core/core/src/engine.rs#L955 
and fail if the conversion of all articles failed: 
https://github.com/xaynetwork/xayn_discovery_engine/blob/126a0b034d5c7aa0ade8d30d6e75f108fda96933/discovery_engine_core/core/src/engine.rs#L655-L658

But since `impl TryFrom<(GenericArticle, StackId, Embedding)> for Document` can't fail anymore the conversion errors won't be logged and collected + we/team blue can't differentiate anymore if the response was empty or faulty. 

Maybe we should address in the next daily because I'm not sure if that's fine or not.

Reply:
Fair. But even if this was a problem, and Blue needs to know the difference between empty and faulty responses, then we could provide this via `query_articles`, which can also return an error. IMO the fact that we're currently checking syntactical correctness outside of the provider is coincidental, because we were using the newscatcher specific representation outside of the `providers` crate.

Reply:
But I do agree that this could be worth bringing up in the daily. Unfortunately that also delays it to Tuesday, but well 🤷🏼‍♂️

Reply:
I'm not too worried about blue, but we do that do try to keep as much as possible from each request so we don't want to fail if one document is not valid. We should do the same here, log a fail using tracing and we fail only if everything fails.